dash-pipeline/SAI/src/dashsai.cpp

```diff
@@ -1043,6 +1049,169 @@ sai_status_t DashSai::bulk_remove_objects(
     return agg_status;
 }
 
+sai_status_t DashSai::create(
+        _In_ const P4MetaTable &meta_table,
+        _In_ sai_object_type_t objectType,
+        _Out_ sai_object_id_t* objectId,
+        _In_ sai_object_id_t switchId,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list)
+{
+    DASH_LOG_ENTER();
+    DASH_CHECK_API_INITIALIZED();
+
+    auto attrs = DashSai::populateDefaultAttributes(objectType, attr_count, attr_list);
+    attr_count = (uint32_t)attrs.size();
+    attr_list = attrs.data();
+
+    std::shared_ptr<p4::v1::TableEntry> matchActionEntry;
+    matchActionEntry = std::make_shared<p4::v1::TableEntry>();
+    matchActionEntry->set_table_id(meta_table.id);
+
+    sai_object_id_t objId = getNextObjectId(objectType);
+    if (objId == SAI_NULL_OBJECT_ID)
+    {
+        DASH_LOG_ERROR("getNextObjectId failed for OBJECT_TYPE %u", objectType);
+        return SAI_STATUS_FAILURE;
+    }
+
+    if (auto meta_object_key = meta_table.get_meta_object_key())
+    {
+        auto key_mf = matchActionEntry->add_match();
+        auto key_mf_exact = key_mf->mutable_exact();
+
+        key_mf->set_field_id(meta_object_key->id);
+        u16SetVal((uint16_t)objId, key_mf_exact, 16);
+    }
+
+    pi_p4_id_t action_id = meta_table.find_action_id(attr_count, attr_list);
+    if (!action_id)
+    {
+        DASH_LOG_ERROR("Not find p4 table action");
+        return SAI_STATUS_FAILURE;
+    }
+
+    auto action = matchActionEntry->mutable_action()->mutable_action();
+    action->set_action_id(action_id);
+
+    for (uint32_t i = 0; i < attr_count; i++)
+    {
+        if (auto meta_param = meta_table.get_meta_action_param(action_id, attr_list[i].id))
```

Comment:
will be better to move the assignment out of the if. this code pattern is easy to go wrong in general.

Reply:
fixed